src/czar/CzarFamilyMap.cc

```diff
@@ -0,0 +1,263 @@
+        qmeta::QMetaChunkMap const& qChunkMap, bool usingChunkSize) {
+    // Create new maps.
+    std::shared_ptr<FamilyMapType> newFamilyMap = make_shared<FamilyMapType>();
+
+    // Workers -> Databases map
+    LOGS(_log, LOG_LVL_DEBUG, cName(__func__) << " workers.sz=" << qChunkMap.workers.size());
+    for (auto const& [workerId, dbs] : qChunkMap.workers) {
+        // Databases -> Tables map
+        for (auto const& [dbName, tables] : dbs) {
+            // Tables -> Chunks map
+            for (auto const& [tableName, chunks] : tables) {
+                // vector of ChunkInfo
+                for (qmeta::QMetaChunkMap::ChunkInfo const& chunkInfo : chunks) {
+                    try {
+                        int64_t chunkNum = chunkInfo.chunk;
+                        CzarChunkMap::SizeT sz = 1;
+                        if (usingChunkSize) {
+                            sz = chunkInfo.size;
+                        }
+                        LOGS(_log, LOG_LVL_DEBUG,
+                             cName(__func__) << "workerdId=" << workerId << " db=" << dbName << " table="
+                                             << tableName << " chunk=" << chunkNum << " sz=" << sz);
+                        _insertIntoMaps(newFamilyMap, workerId, dbName, tableName, chunkNum, sz);
+                    } catch (invalid_argument const& exc) {
+                        throw ChunkMapException(
+                                ERR_LOC, cName(__func__) + " invalid_argument workerdId=" + workerId +
+                                                 " db=" + dbName + " table=" + tableName +
+                                                 " chunk=" + to_string(chunkInfo.chunk) + " " + exc.what());
+                    } catch (out_of_range const& exc) {
+                        throw ChunkMapException(
+                                ERR_LOC, cName(__func__) + " out_of_range workerdId=" + workerId +
+                                                 " db=" + dbName + " table=" + tableName +
+                                                 " chunk=" + to_string(chunkInfo.chunk) + " " + exc.what());
+                    }
+                }
+            }
+        }
+    }
+
+    // This needs to be done for each CzarChunkMap in the family map.
+    for (auto&& [familyName, chunkMapPtr] : *newFamilyMap) {
+        LOGS(_log, LOG_LVL_DEBUG, cName(__func__) << " working on " << familyName);
+        auto missing = chunkMapPtr->organize();
+        if (missing != nullptr && !missing->empty()) {
+            // TODO:DM-53240 Some element of the dashboard should be made aware of this. Also,
+            // TODO:DM-53239 maybe this should check all families before throwing.
+            //             There are implications that maybe the replicator should not
+            //             tell the czar about families/databases that do not have
+            //             at least one copy of each chunk with data loaded on a worker.
```

Comment:
Noted. Though I don't quite understand what the actual problem is here. We could discuss this beyond this PR.